libvast/src/system/importer.cpp

```diff
@@ -38,6 +38,79 @@
 
 namespace vast::system {
 
+namespace {
+
+class driver : public importer_state::driver_base {
+public:
+  using super = importer_state::driver_base;
+
+  using pointer = importer_actor*;
+
+  driver(importer_state::downstream_manager& out, pointer self)
+    : super(out), self_(self) {
+    // nop
+  }
+
+  void process(caf::downstream<output_type>& out,
+               std::vector<input_type>& xs) override {
+    VAST_TRACE(VAST_ARG(xs));
+    auto& st = self_->state;
+    uint64_t events = 0;
+    auto t = timer::start(st.measurement_);
+    for (auto& x : xs) {
+      VAST_ASSERT(x.rows() <= static_cast<size_t>(st.available_ids()));
+      auto rows = x.rows();
+      events += rows;
+      x.offset(st.next_id(rows));
+      out.push(std::move(x));
+    }
+    t.stop(events);
```

Comment:
This PR should not change the measurement logic. The old code ran the timer per table slice, this now runs it once per vector of table slices.

It also doesn't hurt to name the input and output type here explicitly, I think, and to remove them from the `importer_state`, where they shouldn't have been in for a while now.

Reply:
I don't understand, the duration outside of the loop should be the same as the sum of the durations of the loop body?

And you mean the `using` decls? Shure that can be cleaned up.